CRM/Admin/Form/Setting/Debugging.php

```diff
@@ -54,4 +55,17 @@ public function buildQuickForm() {
     parent::buildQuickForm();
   }
 
+  /**
+   * Process the form submission.
+   */
+  public function postProcess() {
+    $params = $this->controller->exportValues($this->_name);
+
+    if ($params['isProductionEnvironment'] === FALSE) {
```

Comment:
I mean - I wasn't expecting the call backs relevant to this setting to be doing any thing in the form layer. ie. this is setting a setting in the form post process hook - if you look at the logging setting, by contrast, disabling & enabling the field has a callback that does the needed, regardless where it is called from